[Flutter] Add documentation for different routing packages (#18721)
DataDog/documentation: content/en/real_user_monitoring/flutter/_index.md

This works if you are using named routes or if you have supplied a name to the `settings` parameter of your `PageRoute`.

If you are not using named routes, you can use `DatadogRouteAwareMixin` in conjunction with the `DatadogNavigationObserverProvider` widget to start and stop your RUM views automatically. With `DatadogRouteAwareMixin`, move any logic from `initState` to `didPush`.

### Flutter Navigator v2

> If you are using Flutter Navigator v2.0, which uses the `MaterialApp.router` named constructor, the setup will vary based on the routing middleware you are using, if any. Since [go_router][11], uses the same observer interface as Flutter Navigator v1, so the `DatadogNavigationObserver` can be added to other observers as a parameter to `GoRouter`.

Reviewer: ```suggestion
If you are using Flutter Navigator v2.0, which uses the `MaterialApp.router` named constructor, the setup varies based on the routing middleware you are using, if any. Since [go_router][11], uses the same observer interface as Flutter Navigator v1, so the `DatadogNavigationObserver` can be added to other observers as a parameter to `GoRouter`.
```

```dart
final _router = GoRouter(
  routes: [
    // Your route information here
  ],
  observers: [
    DatadogNavigationObserver(datadogSdk: DatadogSdk.instance),
  ],
);
MaterialApp.router(
  routerConfig: _router,
  // Your remaining setup
)
```

> For examples that use routers other than `go_router` see Advanced Configuration - Automatic View Tracking

Reviewer: ```suggestion
For examples that use routers other than `go_router`, see [Advanced Configuration - Automatic View Tracking][12].
```

[8]: https://pub.dev/documentation/datadog_flutter_plugin/latest/datadog_flutter_plugin/ViewInfoExtractor.html
[9]: https://api.flutter.dev/flutter/dart-io/HttpOverrides/current.html
[10]: https://pub.dev/documentation/datadog_tracking_http_client/latest/datadog_tracking_http_client/DatadogTrackingHttpOverrides-class.html

> [11]: https://pub.dev/packages/go_router

Reviewer: ```suggestion
[11]: https://pub.dev/packages/go_router
[12]: /real_user_monitoring/flutter/advanced_configuration/#automatic-view-tracking
```
